app/services/coupons/create_service.rb

```diff
@@ -31,18 +45,40 @@ def create(args)
 
     private
 
+    attr_reader :limitations, :organization_id
+
     def track_coupon_created(coupon)
       SegmentTrackJob.perform_later(
         membership_id: CurrentContext.membership,
         event: 'coupon_created',
         properties: {
           coupon_code: coupon.code,
           coupon_name: coupon.name,
-          organization_id: coupon.organization_id,
+          organization_id:,
         },
       )
     end
 
+    def plan_identifiers
```

Comment:
Just a suggestion to make it a bit less verbose:
```ruby
def plan_identifiers
  key = api_context? ? :plan_code : :id
  limitations[key]&.compact&.uniq
end

def plans
  return @plans if defined?(@plans)
  return [] unless plan_identifiers.present?
  
  finder = api_context? ? :code : :id  
  Plan.where(finder => identierier, organization_id:)
end
```

Reply:
refactored, thanks!